clients/roscpp/include/ros/callback_queue_interface.h

```diff
@@ -72,6 +73,7 @@ class ROSCPP_DECL CallbackInterface
   virtual bool ready() { return true; }
 };
 typedef boost::shared_ptr<CallbackInterface> CallbackInterfacePtr;
+typedef boost::weak_ptr<CallbackInterface> CallbackInterfaceWPtr;
```

Comment:
Can you please rename the typedef to be a little bit more verbose:

    CallbackInterfaceWeakPtr